Tests/AK/TestAKMath.cpp

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2023, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/Macros.h>
+#include <LibTest/TestCase.h>
+
+#include <AK/Array.h>
+#include <AK/Math.h>
+
+struct MathTest {
+    double input;
+    double expected_output;
+};
+
+// Test cases for pow(2, x) generated with Python + numpy:
+// echo -e "import numpy as np\nfor x in np.arange(-1020, 1020, 0.9):\n    print(f'{{ {x}, {2**x} }},')" | python3
+// FIXME: Actually use these for a test of AK::pow; currently many inputs are broken due to our bad implementation.
+TEST_CASE(pow)
+{
+    EXPECT_EQ(AK::pow(10., 0.), 1.);
+    EXPECT_EQ(AK::pow(10., 1.), 10.);
+    EXPECT_EQ(AK::pow(10., 2.), 100.);
+    EXPECT_EQ(AK::pow(10., 3.), 1'000.);
+    EXPECT_EQ(AK::pow(10., 4.), 10'000.);
+    EXPECT_EQ(AK::pow(10., 5.), 100'000.);
+    EXPECT_EQ(AK::pow(10., 6.), 1'000'000.);
+    EXPECT_EQ(AK::pow(1.e18, 2.), 1.e36);
+    EXPECT_EQ(AK::pow(1.e19, 2.), 1.e38);
+    // Should not take the integer fast path anymore, since we're beyond 64-bit integers.
```

Comment:
Your change in `AK::pow` relates to the `y` argument, which is `2.` in these tests. How would that exceed a 64-bit integer?

Reply:
The result exceeds 64 bits, not the arguments.

Reply:
Your comment states "should not take the integer fast path", but that has nothing to do with the result being beyond a 64-bit integer. The fast path seems to be dependent on the `y` argument being equal to itself casted to an `i64` and fitting withing the `i64` limits.

Reply:
Just to be completely clear, these are the conditions for the fast path:

```cpp
    if (y >= static_cast<T>(NumericLimits<i64>::min()) && y <= static_cast<T>(NumericLimits<i64>::max())) {
        i64 y_as_int = static_cast<i64>(y);
        if (y == static_cast<T>(y_as_int)) {
```

So having `y == 2.` does not test the absence of a fast path afaics.